### PR TITLE
#162265996 Reusable navbar

### PR DESCRIPTION
**Pull request by victor-nweze-deel**
#### What does this PR do?
- *Add reusable nav bar*

#### Description of Task to be completed?
- Create dynamic and reusable navigation bar
- Add CSS styling to navigation bar

#### How should this be manually tested?
Visit [*Authors Haven*](https://haven-ah-frontend.herokuapp.com/) to view the production build
Or
- git clone this repo
- Cd into the `haven-ah-frontend` folder
- Run `npm run start:dev`

#### What are the relevant pivotal tracker stories?
[#162265996](https://www.pivotaltracker.com/story/show/162265996)

#### Screenshots (if appropriate)
screen shot 2018-11-28 at 1 46 13 pm
screen shot 2018-11-28 at 1 45 53 pm

### CODE REVIEW

**Comment by ucheg6** [comment]
Good work @ProfJigsaw

File: src/components/containers/navbar.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+
+const NavBar = ({ isLoggedIn }) => {
+  return (
+    <div className="layout">
+
+      <nav className="navbar general__nav">
+        <div className="container">
+          <div className="navbar-brand">
+            <a className="navbar-item" href="#">
```
Comment (TheoOkafor) [inline src/components/containers/navbar.js]: React-Router `Link` can be used here in place of `a` tags since we have the react-router already.

**Review by TheoOkafor — COMMENTED** [review]
Looks Good 👍

File: src/components/containers/navbar.js (hunk 1 above)
Comment (victor-nweze-deel) [reply src/components/containers/navbar.js]: Nice catch. Thanks Theo

File: public/index.html [hunk 2]
```diff
@@ -6,6 +6,9 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <meta http-equiv="X-UA-Compatible" content="ie=edge">
     <title>Authors Haven</title>
+    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.5.0/css/all.css" integrity="sha384-B4dIYHKNBt8Bc12p+WXckhzcICo0wtJAoU8YZTY5qE0Id1GSseTk6S+L3BlXeVIU"
```
Comment (seyi-adeleke) [inline public/index.html]: please install font awesome and bulma from npm

File: public/styles/App.scss [hunk 3]
```diff
@@ -1,15 +1,9 @@
-$color: red;
-
-@mixin my-border($color) {
-  border: 1px solid $color;
+.general__nav {
```
Comment (seyi-adeleke) [inline public/styles/App.scss]: this isn't following BEM

Comment (seyi-adeleke) [reply public/styles/App.scss]: Lets import component styles into this file. Each component should have its own stylesheet

File: public/styles/App.scss [hunk 4]
```diff
@@ -1,15 +1,9 @@
...
-h1 {
-    color: #27aedb;
-    text-align: center;
-}
-
-body {
-  @include my-border(green);
-  background: $color;
+.navbar-brand > .navbar-item {
+  color:#777;
```
Comment (seyi-adeleke) [inline public/styles/App.scss]: make this color a variable

File: src/components/HomePage.js [hunk 5]
```diff
@@ -1,13 +1,19 @@
 import React, { PureComponent } from 'react';
```
Comment (seyi-adeleke) [inline src/components/HomePage.js]: All components should have a `.jsx` extension

File: src/components/containers/navbar.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+import propTypes from 'prop-types';
+import CommonNavItems from '../views/commonNavItems';
+import UserNavItems from '../views/userNavItems';
+import GuestNavItems from '../views/guestNavItems';
+
+
+const NavBar = ({ isLoggedIn }) => {
+  return (
+    <div className="layout">
+
```
Comment (seyi-adeleke) [inline src/components/containers/navbar.js]: remove excess spaces

File: src/components/containers/navbar.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import propTypes from 'prop-types';
+import CommonNavItems from '../views/commonNavItems';
+import UserNavItems from '../views/userNavItems';
+import GuestNavItems from '../views/guestNavItems';
+
+
+const NavBar = ({ isLoggedIn }) => {
```
Comment (seyi-adeleke) [inline src/components/containers/navbar.js]: Lets use React.memo here

File: src/components/views/commonNavItems.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+    <div className="navbar-start">
+      <Link className="navbar-item" to="/">
+      Home
+      </Link>
+      <div className="navbar-item has-dropdown is-hoverable">
+        <Link className="navbar-link" to="#">
+          Tags
+        </Link>
+        <div className="navbar-dropdown is-boxed">
+          <Link className="navbar-item" to="#">
```
Comment (seyi-adeleke) [inline src/components/views/commonNavItems.js]: lets have these in an array and Loop over it

**Review by seyi-adeleke — CHANGES_REQUESTED** [review]

**Review by seyi-adeleke — APPROVED** [review]

**Pull request merged**